assembly/src/parser/mod.rs

```diff
@@ -159,6 +159,176 @@ fn parse_forms_internal(
         .map_err(|err| ParsingError::from_parse_error(source_id, err))
 }
 
+// DIRECTORY PARSER
+// ================================================================================================
+
+/// Read the contents (modules) of this library from `dir`, returning any errors that occur
+/// while traversing the file system.
+///
+/// Errors may also be returned if traversal discovers issues with the modules, such as
+/// invalid names, etc.
+///
+/// Returns an iterator over all parsed modules.
+#[cfg(feature = "std")]
+pub fn read_modules_from_dir(
```

Comment:
*Nit:* Since the `SourceManager` here is never passed as an `Arc` anywhere, you could just pass it by reference, i.e. `&dyn SourceManager`. Then in places like `Assembler::add_modules_from_dir`, you can avoid the unnecessary `.clone()`.

Reply:
Agreed! Fixed.